### PR TITLE
ENGESC-5953 remove empty stack versions when gathering stack versions

### PR DESCRIPTION
**Pull request by sodre90**
See detailed description in the commit message.

### CODE REVIEW

File: core/src/main/java/com/sequenceiq/cloudbreak/service/cluster/ambari/InstanceMetadataUpdater.java [hunk 1]
```diff
@@ -264,12 +265,24 @@ private void addPackageVersionsFromGrain(GatewayConfig gatewayConfig, HostOrches
         return packageVersionsByNameByHost;
     }
 
+    private void removeEmptyStack(ArrayList<String> stackTypedPackages, Entry<String, Map<String, String>> entry) {
+        if (!CollectionUtils.isEmpty(stackTypedPackages)) {
+            Set<String> keySet = entry.getValue().keySet();
+            for (String stackTypedPackage : stackTypedPackages) {
+                if (keySet.contains(stackTypedPackage)) {
+                    if (StringUtils.isEmpty(entry.getValue().get(stackTypedPackage))) {
+                        entry.getValue().remove(stackTypedPackage);
```
Comment (pdarvasi) [inline core/src/main/java/com/sequenceiq/cloudbreak/service/cluster/ambari/InstanceMetadataUpdater.java:274]: a log line would be nice here

File: core/src/main/java/com/sequenceiq/cloudbreak/service/cluster/ambari/InstanceMetadataUpdater.java [hunk 2]
```diff
@@ -264,12 +265,24 @@ private void addPackageVersionsFromGrain(GatewayConfig gatewayConfig, HostOrches
         return packageVersionsByNameByHost;
     }
 
+    private void removeEmptyStack(ArrayList<String> stackTypedPackages, Entry<String, Map<String, String>> entry) {
```
Comment (pdarvasi) [inline core/src/main/java/com/sequenceiq/cloudbreak/service/cluster/ambari/InstanceMetadataUpdater.java:268]: a unit test covering this would be good

**Review by pdarvasi — APPROVED** [review]

File: core/src/main/java/com/sequenceiq/cloudbreak/service/cluster/ambari/InstanceMetadataUpdater.java (hunk 2 above)
Comment (sodre90) [reply core/src/main/java/com/sequenceiq/cloudbreak/service/cluster/ambari/InstanceMetadataUpdater.java:268]: done

File: core/src/main/java/com/sequenceiq/cloudbreak/service/cluster/ambari/InstanceMetadataUpdater.java (hunk 1 above)
Comment (sodre90) [reply core/src/main/java/com/sequenceiq/cloudbreak/service/cluster/ambari/InstanceMetadataUpdater.java:274]: done

**Pull request merged**